### PR TITLE
misc updates

### PR DESCRIPTION
**Pull request by lifenxiang**

### CODE REVIEW

File: www/carrier.ts [hunk 1]
```diff
@@ -437,19 +442,16 @@ class CarrierImpl implements CarrierPlugin.Carrier {
         var me = this;
         var _onSuccess = function(ret){
             delete me.carrierManager.groups[group.groupId];
```
Comment (stiartsly) [inline www/carrier.ts]: Why me.carrierManager also manage the group management besides carrier object itself ?

File: www/carrier.ts [hunk 2]
```diff
@@ -437,19 +442,16 @@ class CarrierImpl implements CarrierPlugin.Carrier {
...
     }
 
-    getGroups(onSuccess: (groups: CarrierPlugin.Group[]) => void, onError?: (err: string) => void) {
-        var groups = [];
-        var index = 0;
-        for(var i in this.carrierManager.groups) {
-            groups[index] = this.carrierManager.groups[i];
-            index = index+1;
-        }
+    getGroups(onSuccess: (groups: {[id: string]: CarrierPlugin.Group}) => void, onError?: (err: string) => void) {
```
Comment (stiartsly) [inline www/carrier.ts]: What is the meanings about parameters: groups: {[id: string]: CarrierPlugin.Group} ???

File: www/types.d.ts [hunk 3]
```diff
@@ -824,7 +824,7 @@ declare namespace CarrierPlugin {
          * @param onSuccess  The function to call when success.The param is a group array object ,
          * @param onError    The function to call when error, the param is a string. Or set to null.
          */
-        getGroups(onSuccess:(groups: Group[])=>void, onError?:(err: string)=>void);
+        getGroups(onSuccess:(groups: {[id: string]: Group})=>void, onError?:(err: string)=>void);
```
Comment (stiartsly) [inline www/types.d.ts]: Don't get this one ?

File: www/carrier.ts (hunk 1 above)
Comment (lifenxiang) [reply www/carrier.ts]: It has been always working this way, it didn't change anything.

File: www/types.d.ts [hunk 4]
```diff
@@ -626,7 +626,7 @@ declare namespace CarrierPlugin {
         * @param fileInfo    Information of the file which the requester wants to send
         */
         onConnectRequest?(carrier: Carrier, from: string, fileInfo: FileTransferInfo);
-    }
+    } & GroupCallbacks;
```
Comment (stiartsly) [inline www/types.d.ts]: Why need &GroupCallbacks?

File: www/carrier.ts (hunk 1 above)
Comment (stiartsly) [reply www/carrier.ts]: I think you need change this one. Only one group management would be allowed within carrier instance.

File: www/carrier.ts (hunk 2 above)
Comment (lifenxiang) [reply www/carrier.ts]: groups is an object whose string properties represent group ID and values of string properties are of type CarrierPlugin.Group

Comment (stiartsly) [reply www/carrier.ts]: GroupID don't have to be known by users. User should only care about group object, and then use group object to fetch all it's properties, including it's group ID (but still not recommend to export group ID to users).

File: www/types.d.ts (hunk 3 above)
Comment (stiartsly) [reply www/types.d.ts]: should keep this prototype:  getGroups(onSuccess:(groups: Group[])=>void, onError?:(err: string)=>void);

File: src/android/CarrierPlugin.java [hunk 5]
```diff
@@ -404,6 +404,15 @@ private void createObject(JSONArray args, CallbackContext callbackContext) throw
               r.put("address", carrierHandler.mCarrier.getAddress());
               r.put("nospam", carrierHandler.mCarrier.getNospam());
               r.put("presence", carrierHandler.mCarrier.getPresence().value());
+              JSONArray a = new JSONArray();
+              for (Group group: carrierHandler.mCarrier.getGroups()) {
+                  String groupId = getGroupId();
+                  addGroupMap(groupId, group);
+                  carrierHandler.groups.put(group, groupId);
```
Comment (stiartsly) [inline src/android/CarrierPlugin.java]: carrierHandler also need group management?

**Comment by stiartsly** [comment]
Deprecated and closed.

**Pull request closed without merging**